app/imgur.js

```diff
@@ -0,0 +1,157 @@
+var request = require('request');
+var $ = require('jquery-deferred');
+var util = require('./util');
+
+/**
+ * a simple module for searching Imgur.com for images
+ * or fetching the images from a public album.
+ * see https://api.imgur.com/models/image
+ * see https://api.imgur.com/models/album
+ * @param {string} apiKey, your app's Imgur API key
+ * @constructor
+ */
+function Imgur (apiKey) {
+    this.apiKey = apiKey;
+}
+
+/**
+ * see https://api.imgur.com/models/image
+ * Searches Imgur.com for images returning a promise.
+ * The promise resolves with an array of objects
+ * If no images are found, promise resolves with an empty array.
+ * Promise rejects only on service error.
+ * @param {string} query (append "+ext:gif" for Gifs!)
```

Comment:
Capitalize the parameters between { } - {String} / {Number}. Use proper sentence notation.